File: Keyword-Spring/src/main/java/com/keyword/keywordspring/model/Comment.java
package com.keyword.keywordspring.model;

import lombok.*;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Date;

@Entity
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Comment implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Lob
    private String content;

    @ManyToOne
    private AppUser user;

    @ManyToOne
    private Post post;

    @OneToOne
    private Comment parentComment;

    private Date dateCreated;

    private Boolean edited;

    private Integer votes;
}

File: Keyword-Angular/src/app/model/comment.ts
import { VoteType } from './voteType';
export class Comment{
    id: number = 0;
    content: string = "content";
    parentCommentId: number | null = null;
    user: string = "username";
    postId: number = 0;
    dateCreated: Date = new Date();
    votes: number = 0;
    userVote: VoteType | null = null;
}